crates/ra_parser/src/grammar/patterns.rs

```diff
@@ -58,6 +58,12 @@ const PAT_RECOVERY_SET: TokenSet =
 fn atom_pat(p: &mut Parser, recovery_set: TokenSet) -> Option<CompletedMarker> {
     let la0 = p.nth(0);
     let la1 = p.nth(1);
+    if la0 == IDENT && la1 == T![..] {
```

Comment:
Hm, I think this would allow `x..` anywhere where patterns are allowed, and not only inside slices. I think we need to adjust `slice_pat` instead.

Reply:
Ahh you're right of course. I inlined `pat_list()` to `slice_pat()` and made the adjustment there. Seeing as `pat_list()` now only had one caller, I inlined it there and deleted it.